src/types/weakref/proxy.rs

```diff
@@ -14,22 +14,31 @@ use super::PyWeakrefMethods;
 #[repr(transparent)]
 pub struct PyWeakrefProxy(PyAny);
 
-pyobject_native_type_named!(PyWeakrefProxy);
+pyobject_native_type_core!(
+    PyWeakrefProxy,
+    |py| {
+        static TYPE: PyOnceLock<Py<PyType>> = PyOnceLock::new();
+        TYPE.get_or_try_init(py, || {
+            // Terrible way to generate a super class of both ProxyType and CallableProxyType
+            let globals = PyDict::new(py);
+            globals.set_item("abc", py.import("abc")?)?;
+            py.run(ffi::c_str!("class WeakrefProxy(abc.ABC):\n    pass"), Some(&globals), None)?;
+            let cls = globals.get_item("WeakrefProxy")?.downcast_into::<PyType>()?;
+            let weakref = py.import("weakref")?;
+            for child_name in ["ProxyType", "CallableProxyType"] {
+                cls.call_method1("register", (weakref.getattr(child_name)?,))?;
+            }
+            PyResult::Ok(cls.unbind())
+        }).unwrap().bind(py).as_type_ptr()
```

Comment:
Here I wonder if we could use `weakref.ProxyTypes`. That might require logic to allow the "type object" to not be a real type object but also a tuple (or union)? That would break other behaviour about `PyTypeInfo`, however.